web3.js/src/utils/cluster.ts

```diff
@@ -2,22 +2,24 @@ const endpoint = {
   http: {
     devnet: 'http://api.devnet.solana.com',
     testnet: 'http://api.testnet.solana.com',
-    'mainnet-beta': 'http://api.mainnet-beta.solana.com/',
+    'mainnet-beta': 'http://api.mainnet-beta.solana.com',
+    localnet: 'http://localhost:8899'
   },
   https: {
     devnet: 'https://api.devnet.solana.com',
     testnet: 'https://api.testnet.solana.com',
-    'mainnet-beta': 'https://api.mainnet-beta.solana.com/',
+    'mainnet-beta': 'https://api.mainnet-beta.solana.com',
+    localnet: 'https://localhost:8899'
   },
 };
 
-export type Cluster = 'devnet' | 'testnet' | 'mainnet-beta';
+export type Cluster = 'devnet' | 'testnet' | 'mainnet-beta' | 'localnet';
 
 /**
  * Retrieves the RPC API URL for the specified cluster
  */
 export function clusterApiUrl(cluster?: Cluster, tls?: boolean): string {
-  const key = tls === false ? 'http' : 'https';
+  const key = tls === false || (tls == undefined && cluster === 'localnet') ? 'http' : 'https';
```

Comment:
If `tls` is true or undefined, for all clusters except localnet, default to https.
But for localnet, default to http unless it is explicitly true.

Reply:
Isn't it the opposite – localhost always presumed to be secure? https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts#when_is_a_context_considered_secure

Reply:
Oh I see. This is the URL getter.